src/uwtools/cli.py

```diff
@@ -305,6 +307,99 @@ def _dispatch_forecast_run(args: Namespace) -> bool:
     )
 
 
+# Mode Rocoto
+
+
+def _add_subparser_rocoto(subparsers: Subparsers) -> ModeChecks:
+    """
+    Subparser for mode: rocoto
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.rocoto, "Realize and validate Rocoto XML Documents")
+    _basic_setup(parser)
+    subparsers = _add_subparsers(parser, STR.submode)
+    return {
+        STR.realize: _add_subparser_rocoto_realize(subparsers),
+        STR.validate: _add_subparser_rocoto_validate(subparsers),
+    }
+
+
+def _add_subparser_rocoto_realize(subparsers: Subparsers) -> SubmodeChecks:
+    """
+    Subparser for mode: rocoto realize
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.realize, "Realize a Rocoto XML workflow document")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_input_file(required)
+    _add_arg_output_file(required)
+    optional = _basic_setup(parser)
+    checks = _add_args_quiet_and_verbose(optional)
+    return checks + [
+        partial(_check_file_vs_format, STR.infile, FORMAT.yaml),
+        partial(_check_file_vs_format, STR.outfile, FORMAT.rocoto),
```

Comment:
I'm not sure either of these checks is necessary. This check seems to only make sure that when the user doesn't specify a file format, that it can be derived from the file name.